src/plugins/validation/2and3/semantic-validators/schema-ibm.js

```diff
@@ -139,60 +144,135 @@ function generateFormatErrors(schema, contextPath, config, isOAS3, messages) {
+          )}.`,
+          checkStatus
+        );
+      }
       break;
     case 'number':
-      valid =
-        !property.format ||
-        includes(['float', 'double'], property.format.toLowerCase());
+      if (
+        obj.format &&
+        !includes(validNumberFormats, obj.format.toLowerCase())
+      ) {
+        messages.addMessage(
+          path.concat(['type']),
+          `Schema of type number should use one of the following formats: ${validNumberFormats.join(
+            ', '
+          )}.`,
+          checkStatus
+        );
+      }
       break;
     case 'string':
-      valid =
-        !property.format ||
-        includes(
-          ['byte', 'binary', 'date', 'date-time', 'password'],
-          property.format.toLowerCase()
+      if (
+        obj.format &&
+        !includes(validStringFormats, obj.format.toLowerCase())
+      ) {
+        messages.addMessage(
+          path.concat(['type']),
+          `Schema of type string should use one of the following formats: ${validStringFormats.join(
+            ', '
+          )}.`,
+          checkStatus
         );
+      }
       break;
     case 'boolean':
-      valid = property.format === undefined; // No valid formats for boolean -- should be omitted
+      // No valid formats for boolean, format should be undefined
+      if (obj.format !== undefined) {
+        messages.addMessage(
+          path.concat(['type']),
+          `Schema of type boolean should not have a format.`,
+          checkStatus
+        );
+      }
       break;
     case 'object':
+      // No valid format pairings for object, format should be undefined
+      if (obj.format !== undefined) {
+        messages.addMessage(
+          path.concat(['type']),
+          'Schema of type object should not have a format.',
+          checkStatus
+        );
+      }
+      break;
     case 'array':
-      valid = true;
+      // No valid format pairings for array, format should be undefined
+      if (obj.format !== undefined) {
+        messages.addMessage(
+          path.concat(['type']),
+          'Schema of type array should not have a format.',
+          checkStatus
+        );
+      }
       break;
     case 'file':
       // schemas of type file are allowed in swagger2 for responses and parameters
-      // of type 'formData' - the violating parameters are caught by parameters-ibm
+      // of type 'formData'
       // note: type file is only allowed for root schemas (not properties, etc.)
-      valid = !isOAS3 && isRootSchema(path);
+      if (isOAS3 || (!obj.in && !isRootSchema(path))) {
+        messages.addMessage(
+          path.concat(['type']),
+          'File type only valid for swagger2 and must be used as root schema.',
+          checkStatus
+        );
+      } else if (obj.in && obj.in !== 'formData') {
+        messages.addMessage(
+          path.concat(['type']),
+          'File type parameter must use in: formData.',
+          checkStatus
+        );
+      }
```

Comment:
This may not be a big deal since it's likely a small use case, but "format" shouldn't be specified for type "file" Since you check for that for some types above, it might make sense to do that here. Not a requirement

Reply:
Yes, I will add this.